app/gui/src/model/execution_context/plain.rs

```diff
@@ -247,15 +259,17 @@ pub mod test {
 
     use double_representation::definition::DefinitionName;
     use double_representation::project;
+    use engine_protocol::language_server;
 
     #[derive(Clone, Derivative)]
     #[derivative(Debug)]
     pub struct MockData {
-        pub module_path:     model::module::Path,
-        pub context_id:      model::execution_context::Id,
-        pub root_definition: DefinitionName,
-        pub namespace:       String,
-        pub project_name:    String,
+        pub module_path:         model::module::Path,
+        pub context_id:          model::execution_context::Id,
+        pub root_definition:     DefinitionName,
+        pub namespace:           String,
+        pub project_name:        String,
+        pub ls_component_groups: Vec<language_server::LibraryComponentGroup>,
```

Comment:
I was writing about it so many times in your reviews. Please, do not use shortcuts in public struct fields. You can use popular shortcuts in local variables in funcs, but not in publicly available APIs.

Reply:
This could be just `component_groups`, the `language_server` is seen in the type.